portal/migrations/versions/c49b7f7944a7_.py

```diff
@@ -0,0 +1,64 @@
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+
+from portal.models.audit import Audit
+from portal.models.user_consent import UserConsent
+
+"""consent audit & acceptance_date cleanup
+
+Revision ID: c49b7f7944a7
+Revises: 0f1576a4e220
+Create Date: 2018-06-21 14:37:20.145030
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'c49b7f7944a7'
+down_revision = '0f1576a4e220'
+Session = sessionmaker()
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('user_consents', sa.Column(
+        'acceptance_date', sa.DateTime(), nullable=True))
+
+    bind = op.get_bind()
+    session = Session(bind=bind)
+
+    # Whip through every user_consent, migrating data...
+    delete_em = []
+    for uc in session.query(UserConsent):
+        uc.acceptance_date = uc.audit.timestamp
+
+        # look for `recorded` audit that may have different
+        # timestamp
+        comment = "Consent agreement {} recorded".format(
+            uc.id)
+        recorded = session.query(Audit).filter(
+            Audit.comment == comment)
+        recorded = recorded.first()
+        if recorded:
+            assert(uc.audit.user_id == recorded.user_id)
+            assert(uc.audit.context == 'consent')
+            uc.audit.timestamp = recorded.timestamp
+            delete_em.append(recorded)
+
+    # Purge the now obsolete recorded audit rows
```

Comment:
Would it make sense to add an audit record for these changes?

Reply:
i don't think so - all the data we need is still there, simply a better format for storage.